rama-http/src/service/web/endpoint/response/datastar.rs

```diff
@@ -32,3 +33,66 @@ impl IntoResponse for DatastarScript {
         self.0.into_response()
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+/// Datastar source map ready to be served.
+///
+/// Embedded version of the datastar source map file
+/// for improved debugging experience.
+///
+/// Learn more about datastar at <https://data-star.dev/>
+/// or in the book: <https://ramaproxy.org/book/web_servers.html#datastar>.
+pub struct DatastarSourceMap(Script<&'static str>);
```

Comment:
I would store this just as a `&'static str` and within your IntoResponse impl you'll want to do this:

```
(Headers::single(ContentType::json()), self.0).into_response();
```

Something like that, don't know the exact syntax. But the point is that a sourcemap is json, not javascript.